show/stp.py

```diff
@@ -0,0 +1,402 @@
+    app_db_entry = stp_get_all_from_pattern(
+        g_stp_appl_db, g_stp_appl_db.APPL_DB, "*STP_PORT_TABLE:{}".format(ifname))
+    if (not app_db_entry or not ('port_fast' in app_db_entry) or app_db_entry['port_fast'] == 'no'):
+        return False
+    return True
+
+
+def stp_is_uplink_fast_enabled(ifname):
+    entry = g_stp_cfg_db.get_entry("STP_PORT", ifname)
+    if (entry and ('uplink_fast' in entry) and entry['uplink_fast'] == 'true'):
+        return True
+    return False
+
+
+def stp_get_entry_from_vlan_tb(db, vlanid):
+    entry = stp_get_all_from_pattern(db, db.APPL_DB, "*STP_VLAN_TABLE:Vlan{}".format(vlanid))
+    if not entry:
+        return entry
+
+    if 'bridge_id' not in entry:
+        entry['bridge_id'] = 'NA'
+    if 'max_age' not in entry:
+        entry['max_age'] = '0'
+    if 'hello_time' not in entry:
+        entry['hello_time'] = '0'
+    if 'forward_delay' not in entry:
+        entry['forward_delay'] = '0'
+    if 'hold_time' not in entry:
+        entry['hold_time'] = '0'
+    if 'last_topology_change' not in entry:
+        entry['last_topology_change'] = '0'
+    if 'topology_change_count' not in entry:
+        entry['topology_change_count'] = '0'
+    if 'root_bridge_id' not in entry:
+        entry['root_bridge_id'] = 'NA'
+    if 'root_path_cost' not in entry:
+        entry['root_path_cost'] = '0'
+    if 'desig_bridge_id' not in entry:
+        entry['desig_bridge_id'] = 'NA'
+    if 'root_port' not in entry:
+        entry['root_port'] = 'NA'
+    if 'root_max_age' not in entry:
+        entry['root_max_age'] = '0'
+    if 'root_hello_time' not in entry:
+        entry['root_hello_time'] = '0'
+    if 'root_forward_delay' not in entry:
+        entry['root_forward_delay'] = '0'
+    if 'stp_instance' not in entry:
+        entry['stp_instance'] = '65535'
+
+    return entry
+
+
+def stp_get_entry_from_vlan_intf_tb(db, vlanid, ifname):
+    entry = stp_get_all_from_pattern(db, db.APPL_DB, "*STP_VLAN_PORT_TABLE:Vlan{}:{}".format(vlanid, ifname))
+    if not entry:
+        return entry
+
+    if 'port_num' not in entry:
+        entry['port_num'] = 'NA'
+    if 'priority' not in entry:
+        entry['priority'] = '0'
+    if 'path_cost' not in entry:
+        entry['path_cost'] = '0'
+    if 'root_guard' not in entry:
+        entry['root_guard'] = 'NA'
+    if 'bpdu_guard' not in entry:
+        entry['bpdu_guard'] = 'NA'
+    if 'port_state' not in entry:
+        entry['port_state'] = 'NA'
+    if 'desig_cost' not in entry:
+        entry['desig_cost'] = '0'
+    if 'desig_root' not in entry:
+        entry['desig_root'] = 'NA'
+    if 'desig_bridge' not in entry:
+        entry['desig_bridge'] = 'NA'
+
+    return entry
+
+
+#
+# This group houses Spanning_tree commands and subgroups
+@click.group(cls=clicommon.AliasedGroup, invoke_without_command=True)
+@click.pass_context
+def spanning_tree(ctx):
+    """Show spanning_tree commands"""
+    global g_stp_appl_db
```

Comment:
>global g_stp_appl_db [](http://example.com/codeflow?start=4&length=20)

The use of global variables like `g_stp_appl_db`, `g_stp_cfg_db`, `g_stp_vlanid`, and `g_stp_mode` can lead to unexpected behavior, especially in concurrent or complex programs. Consider passing these variables as function parameters or encapsulating them within a class to maintain a cleaner and more modular code structure.